ui/app/components/ui/button/button.component.js

```diff
@@ -25,22 +25,33 @@ const typeHash = {
   'first-time': CLASSNAME_FIRST_TIME,
 }
 
-const Button = ({ type, submit, large, children, icon, rounded, className, ...buttonProps }) => (
-  <button
-    type={submit ? 'submit' : undefined}
-    className={classnames(
-      'button',
-      typeHash[type] || CLASSNAME_DEFAULT,
-      large && CLASSNAME_LARGE,
-      rounded && CLASSNAME_ROUNDED,
-      className,
-    )}
-    { ...buttonProps }
-  >
-    {icon && <span className="button__icon">{icon}</span>}
-    { children }
-  </button>
-)
+const Button = ({ type, submit, large, children, icon, rounded, className, ...buttonProps }) => {
+  // To support using the Button component to render styled links that are semantic html
+  // we swap the html tag we use to render this component and delete any buttonProps that
+  // we know to be erroneous attributes for a link. We will likely want to extract Link
+  // to its own component in the future.
+  let Tag = 'button'
+  buttonProps.type = submit ? 'submit' : undefined
+  if (type === 'link') {
+    Tag = 'a'
+    delete buttonProps.type
+  }
```

Comment:
This made me do a double-take. Can we do this instead?
```suggestion
  if (type === 'link') {
    Tag = 'a'
  } else if (submit) {
    buttonProps.type = 'submit'
  }
```

Reply:
Looks way better ,thanks !